### PR TITLE
refactor: Move loader-agnostic error types from `container-utils` to `telemetry-utils`

### PR DESCRIPTION
**Pull request by Josmithr**
This is part of an effort to remove the container-utils package. The objective is to move the error types in that package to `telemetry-interfaces`. This PR moves all of the loader-agnostic types to their new home.

The moved members are re-exported from `telemetry-interfaces` to maintain backwards compatibility but are marked as deprecated. Repo-internal imports have been updated to import from `telemetry-interfaces`, and dependencies on `container-utils` were removed where possible.

Types moved and their destinations:
* DataCorruptionError: Moved to `telemetry-utils`, re-exported and deprecated in `container-utils` 
* GenericError: Moved to `telemetry-utils`, re-exported and deprecated in `container-utils` 
* UsageError: Moved to `telemetry-utils`, re-exported and deprecated in `container-utils` 
* ThrottlingWarning: Copied to `container-loader` (not exported), deprecated in `container-utils`

### CODE REVIEW

**Comment by Josmithr** [comment]
Note for reviewers: this PR currently leaves 2 API members remaining in `container-utils`: `DataProcessingError` (class) and `extractSafePropertiesFromMessage` (function). These depend on concepts in the `protocol-definitions` package, which `telemetry-utils` cannot see. If we can come to a resolution to this question in this PR, that would be great. But I'm also happy to do that in a separate PR.

**Comment by Josmithr** [comment]
Requires #16868 be merged before this PR can be merged.

File: packages/loader/container-loader/tsconfig.json [hunk 1]
```diff
@@ -6,5 +6,5 @@
 		"outDir": "./dist",
 		"composite": true,
 	},
-	"include": ["src/**/*"],
+	"include": ["src/**/*", "../container-utils/src/error.ts"],
```
Comment (Josmithr) [inline packages/loader/container-loader/tsconfig.json:9]:
```suggestion
	"include": ["src/**/*"],
```

**Review by ghoshkaj — APPROVED** [review]
LGTM! But of course, there's a pesky the merge conflict to resolve first!

File: packages/utils/telemetry-utils/src/error.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	constructor(message: string, readonly error?: any, props?: ITelemetryProperties) {
+		// Don't try to log the inner error
+		super(message, props, new Set(["error"]));
+	}
+}
+
+/**
+ * Warning emitted when requests to storage are being throttled.
+ */
+export class ThrottlingWarning extends LoggingError implements IThrottlingWarning, IFluidErrorBase {
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/error.ts:40]: This one is only used in container-loader, I think we could move it there instead and stop exporting it anywhere.

**Review by alexvy86 — COMMENTED** [review]
One change I think we should make, but otherwise lgtm!

File: packages/utils/telemetry-utils/src/error.ts (hunk 2 above)
Comment (Josmithr) [reply packages/utils/telemetry-utils/src/error.ts:40]: Thanks! Will update!

**Review by anthony-murphy — APPROVED** [review]

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**